src/main/setupDownloadHandlers.js

```diff
@@ -0,0 +1,71 @@
+import path from 'path';
+import fs from 'fs';
+
+import { ipcMain } from 'electron';
+
+import extractZipInplace from './extractZipInplace';
+import { ipcMainChannels } from './ipcMainChannels';
+import { getLogger } from '../logger';
+
+const logger = getLogger(__filename.split('/').slice(-1)[0]);
+
+export default function setupDownloadHandlers(mainWindow) {
```

Comment:
need a docstring here?

Reply:
Thanks, added a docstring. 

> If I'm understanding correctly, it's ok to use the mainWindow object directly here, because this function is called from main.js and so it's run in the main process? Whereas things running in the renderer process need to use IPC?

Yes, I think that's right. Though to be honest I'm not sure what would happen if renderer process function was passed a reference to `mainWindow`. 

Anyway, all of this was previously inside the `createWindow` function in `main.js` where `mainWindow` was already in scope. But that was pretty messy so I wanted a module devoted just to this stuff. But I also needed to use `mainWindow.webContents.send`, so that's why `mainWindow` is passed to the function. I'm not certain that is a good idea though - passing around `mainWindow`. Any thoughts?

Reply:
This seems fine to me